packages/vite/src/node/server/ws.ts

```diff
@@ -66,7 +67,7 @@ export interface WebSocketServer extends NormalizedHotChannel {
   clients: Set<WebSocketClient>
 }
 
-export interface WebSocketClient extends HotChannelClient {
+export interface WebSocketClient extends NormalizedHotChannelClient {
```

Comment:
In this case, I still think it might be better to consider `WebSocketClient` as a normalized client.

Another idea is to have a generic for `NormalizedHotChannelClient<{ socket: ... }>` so that the extra fields can be preserved during normalization.